reana-ui/src/pages/badgeCreator/BadgeLaunchCreator.js

```diff
@@ -0,0 +1,268 @@
+    newParameters[index][key] = value;
+    setLauncherData({ ...launcherData, parameters: newParameters });
+  };
+
+  const handleSubmit = () => {
+    let REANAServerURL = window.location.protocol + "//" + window.location.host;
+    let baseURL = `${REANAServerURL}/launch?`;
+    let errorMessage = null;
+    // Check that the URL is valid
+    try {
+      new URL(launcherData.url);
+    } catch (e) {
+      errorMessage = "The URL is not valid!";
+    }
+    baseURL += `url=${encodeURIComponent(launcherData.url)}`;
+
+    // Validate analysis name
+    if (!errorMessage && launcherData.analysisName) {
+      if (launcherData.analysisName.includes("."))
+        errorMessage = "The analysis name cannot contain dots.";
+      baseURL += `&name=${encodeURIComponent(launcherData.analysisName)}`;
+    }
+
+    // Validate specification file name
+    if (
+      !errorMessage &&
+      launcherData.specFileName &&
+      launcherData.specFileName !== "reana.yaml"
+    ) {
+      if (
+        !launcherData.specFileName.endsWith(".yaml") &&
+        !launcherData.specFileName.endsWith(".yml")
+      ) {
+        errorMessage =
+          "The specification file name must end with .yaml or .yml";
+      }
+      baseURL += `&specification=${encodeURIComponent(
+        launcherData.specFileName,
+      )}`;
+    }
+
+    if (!errorMessage && launcherData.parameters.length > 0) {
+      // The value of each parameter is considered to be a JSON-encoded string.
+      // In this way, the user can pass integers, lists, complex objects, etc.
+      try {
+        const paramsObject = launcherData.parameters.reduce(
+          (acc, param) => ({ ...acc, [param.key]: JSON.parse(param.value) }),
+          {},
+        );
+        const paramsString = JSON.stringify(paramsObject);
+        baseURL += `&parameters=${encodeURIComponent(paramsString)}`;
+      } catch (e) {
+        errorMessage =
+          "The value of the parameters must be a valid JSON string.";
+      }
+    }
+    if (errorMessage) {
+      dispatch(
+        triggerNotification("The form data is invalid!", errorMessage, {
+          error: true,
+        }),
+      );
+      setLauncherURL(null);
+      console.error(errorMessage);
+      return;
+    }
+    setLauncherURL(baseURL);
+  };
+
+  return (
+    <BasePage title="Create your personalized launcher badge">
+      <Container text className={styles["container"]}>
+        <div>
+          <Title>Create your own "Launch on REANA" badge</Title>
+        </div>
+        <p>
+          Fill in the form below to generate the URL to the REANA launcher that
+          will run your analysis, as well as the markdown code for a "Launch on
+          REANA" badge that you can include in your GitHub repositories!
+        </p>
+        <Form onSubmit={handleSubmit}>
+          <Segment className={styles.segment}>
+            <Grid>
+              <Grid.Row>
+                <Grid.Column>
```

Comment:
I think there is no need to use grid in forms, as we can just use a few `Form.Group`? https://react.semantic-ui.com/collections/form/#group-variations-evenly-divided-group

For example:
```jsx
<Form.group widht="equals">
  ...
```

(note that you can also specify the width of a field in "grid columns": https://react.semantic-ui.com/collections/form/#field-variations-width-field )